model/notification.go

```diff
@@ -1,12 +1,146 @@
 package model
 
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+)
+
 type Notification struct {
-	UserID         int    `db:"user_id"`
-	AnnouncementID int    `db:"announcement_id"`
-	Status         string `db:"status"`
-	Receipt        string `db:"receipt"`
-	ReceiptStatus  string `db:"receipt_status"`
-	Timestamp      string `db:"timestamp"`
+	// From the notifications table.
+	ID              string `db:"id"`
+	UserID          int    `db:"user_id"`
+	AnnouncementID  int    `db:"announcement_id"`
+	Status          string `db:"status"`
+	LeaseExpiration int64  `db:"expiration_time"`
+	Receipt         string `db:"receipt"`
+	// From joined tables.
+	ExpoPushToken string `db:"expo_push_token"`
+	Title         string `db:"title"`
+	Body          string `db:"body"`
+}
+
+func EnqueueAnnouncementNotifications(db *sqlx.DB) {
+	log.Println("Starting to enqueue announcement notifications.")
+
+	err := transact(db, func(tx *sqlx.Tx) error {
+		insertQuery := `
+INSERT IGNORE into notifications (user_id, announcement_id, status)
+	SELECT users.id as user_id, announcements.id as announcement_id, "Queued" as status
+	FROM announcements
+	JOIN users ON users.conference_id = announcements.conference_id
+	WHERE sent = 0 AND send_time <= NOW() AND expo_push_token like "ExponentPushToken[%]"
+	ORDER BY send_time asc
+`
+		results := tx.MustExec(insertQuery)
+		notificationRows, err := results.RowsAffected()
+		if err != nil {
+			return fmt.Errorf("failed to insert into notifications table: %w", err)
+		}
+
+		updateQuery := `
```

Comment:
I'm not sure I understand why this query is separate. Maybe add a comment explaining why it's here, and why it's part of the transaction?

Reply:
It's purpose is just to mark the announcement as "sent" in the announcements table, which has two purposes:
- So it can be displayed on the admin UI that the announcement was queued to be sent
- To skip over sent announcements in the first query

I guess it doesn't actually need to be a part of the transaction. It just needs to not run if the first query does not successful complete.